src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexParser.cs

```diff
@@ -107,14 +107,25 @@ public static string Escape(string input)
             {
                 if (IsMetachar(input[i]))
                 {
-                    StringBuilder sb = StringBuilderCache.Acquire();
+                    ReadOnlySpan<char> inputSpan = input.AsSpan();
+
+                    // For small inputs we allocate on the stack. In most cases a buffer three 
+                    // times larger the original string should be sufficient as usually not all 
+                    // characters need to be encoded.
+                    // For larger string we rent the input string's length plus a fixed 
+                    // conservative amount of chars from the ArrayPool.
+                    Span<char> buffer = input.Length <= 80 ? stackalloc char[256] : null;
```

Comment:
This change is completely isolated to the others and perf dropped by 20%. My test strings are always under 80 characters which means stack allocation is used. What could be the cause here?

Reply:
If it is the case, the fix would be to factor the fast path and slow path into separate methods.